app/javascript/react/screens/App/Overview/screens/MappingWizard/MappingWizard.js

```diff
@@ -126,8 +126,18 @@ class MappingWizard extends React.Component {
 
   goToStep(activeStepIndex) {
     const { activeStepIndex: currentStep } = this.state;
+    const targetStepProp = mappingWizardSteps[activeStepIndex];
+    const targetStepForm = this.props[targetStepProp];
 
-    if (currentStep !== 4) {
+    if (activeStepIndex === 4) {
+      this.setState(prevState => ({
+        activeStepIndex: prevState.activeStepIndex
+      }));
+    } else if (currentStep === 2 && activeStepIndex === 3) {
+      this.nextStep();
```

Comment:
We really should be keeping track if the step's status is valid and not allow navigation forward based on the validity. The cursor over following steps in the step indicator should show disabled.